fe/fe-core/src/main/java/com/starrocks/qe/Coordinator.java

```diff
@@ -1087,6 +1121,11 @@ private void computeFragmentHosts() throws Exception {
                 for (int j = 0; j < fragment.getChildren().size(); j++) {
                     int currentChildFragmentParallelism =
                             fragmentExecParamsMap.get(fragment.getChild(j).getFragmentId()).instanceExecParams.size();
+                    // when dop adaptation enabled, numInstances * pipelineDop is equivalent to numInstances in
+                    // non-pipeline engine and pipeline engine(dop adaptation disabled).
+                    if (dopAdaptionEnabled) {
+                        currentChildFragmentParallelism *= fragment.getChild(j).getPipelineDop();
```

Comment:
If dopAdaptionEnabled is true,  ragment.getChild(j).getPipelineDop() will always return 1? so there are no need to compute?

Reply:
For some PlanFragment with ScanNode as left deepmost child, it's pipelineDop is greater than 1.